### ISSUE TITLE
Using an LWRP provider by its name (Ark) directly is no longer supported in Chef 12

### ISSUE THREAD

**Issue by martinb3** [burtlo/ark#117]
Hi there -- I am seeing this warning, and I can't actually tell if it's me or the ark cookbook causing it:
```
Using an LWRP provider by its name (Ark) directly is no longer supported in Chef 12 and will be removed.  Use Chef::ProviderResolver.new(node, resource, action) instead. at 1 location:
           - /tmp/kitchen/cache/cookbooks/ark/resources/default.rb:26:in `initialize'
```

I'm simply using the recipe DSL to make a resource and call `run_action(...)` on it.

**Comment by dwmarshall** [comment]
I think it's the cookbook - our Chef experts have suggested a fix for me, and I'll try to create a pull request for it ASAP.

**Comment by isaac-gittins-amaysim** [comment]
Unfortunately it looks like 0.9.1 is currently failing to build.

**Comment by dwmarshall** [comment]
Indeed, and I'm in communication with @burtlo about how to resolve the build problems. However, if you just want to satisfy yourself that it's the cookbook, you can add the line above to `Berksfile`. Don't use it for anything in production, though!

**Comment by iroller** [comment]
Hi. Any updates here?

**Comment by Puneeth-n** [comment]
+1 Any updates?

**Comment by johntdyer** [comment]
+1

**Comment by vpack** [comment]
+1

**Comment by wfindley** [comment]
+1

**Comment by brodock** [comment]
@isaac-gittins-amaysim @burtlo, this will help getting build working again: https://github.com/burtlo/ark/pull/125. Spec needs to be corrected, but kitchen is working again.

**Comment by dwmarshall** [comment]
This should be fixed by #126

**Comment by tas50** [comment]
Closing this out now that #126 is merged
